### PR TITLE
Step3

### PR DESCRIPTION
**Pull request by yukinkm0**
## What
I have completed up to STEP3.

## CHECKS :warning:

main.py Ln28:  #image.readでcoroutine objectが生成されていたためbinary dataを取り出すためにasyncio.runを追加しました。

Please make sure you are aware of the following.

- [x] **The changes in this PR doesn't have private information

### CODE REVIEW

File: python/items.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{"items": [{"name": "jacket", "category": "fashion"}, {"name": "jacket", "category": "fashion"}, {"name": "jacket", "category": "fashion", "image_filename": "image_filename"}, {"name": "jacket", "category": "fashion", "image_filename": "d5bbb87d9f8dd2c43d6bdc98c5e48d6804c770017643a63ec83986ed987bcfb9.jpg"}, {"name": "jacket", "category": "fashion", "image_filename": "d5bbb87d9f8dd2c43d6bdc98c5e48d6804c770017643a63ec83986ed987bcfb9.jpg"}]}
```
Comment (rhomel) [inline python/items.json:1]: You can add this file to `.gitignore` also.

File: python/main.py [hunk 2]
```diff
@@ -18,17 +21,47 @@
...
+    with open(json_file, mode='r') as j:
+        items = json.load(j)
+    
+    image_filename = store_image(image)
+    
+    if not {'name': name, 'category': category, 'image_name': image_filename} in items['items']:
+        items['items'].append({'name': name, 'category': category, 'image_filename': image_filename})
 
+    with open(json_file, mode='w') as j:
+        json.dump(items, j)
```
Comment (rhomel) [inline python/main.py:56]: Since the item is only added if it does not exist, it is ok to move this to the body of the `if`

File: python/main.py [hunk 3]
```diff
@@ -18,17 +21,47 @@
...
+def add_item(name: str = Form(...), category: str = Form(...), image: UploadFile = File(...)):
+    logger.info(f"Receive item: {name}, {category}")
+
+    with open(json_file, mode='r') as j:
+        items = json.load(j)
+    
+    image_filename = store_image(image)
+    
+    if not {'name': name, 'category': category, 'image_name': image_filename} in items['items']:
+        items['items'].append({'name': name, 'category': category, 'image_filename': image_filename})
```
Comment (rhomel) [inline python/main.py:53]: The specification is not clear here but I think it should be possible to have multiple items that are the same. For example in a book store there will be many copies of the same book.

File: python/main.py [hunk 4]
```diff
@@ -43,3 +76,14 @@ async def get_image(image_name):
         image = images / "default.jpg"
 
     return FileResponse(image)
+
+@app.get("/items/{item_id}")
+def get_item(item_id: int):
+    with open(json_file, mode='r') as j:
+        items = json.load(j)
+
+    if item_id >= len(items['items']):
```
Comment (rhomel) [inline python/main.py:85]: Nice job, but you should also consider if the item id is less than 0 or if the item id is not a number at all.

**Review by rhomel — APPROVED** [review]
Nice job 👍

File: python/items.json (hunk 1 above)
Comment (yukinkm0) [reply python/items.json:1]: Added to  ```.gitignore```.

File: python/main.py (hunk 3 above)
Comment (yukinkm0) [reply python/main.py:53]: Deleted Ln52, which leads to the record being appended directly to the 
```items['items']```.

File: python/main.py (hunk 4 above)
Comment (yukinkm0) [reply python/main.py:85]: STEP4 の際、 SQLite でテーブルを作成する時に  INTEGER PRIMARY KEY をidとして登録するよう指示があったため、現在のようにcodeを書きました。
変更を加えた方がよろしいでしょうか。

**Pull request closed without merging**